app/configurator/components/filters.tsx

```diff
@@ -689,6 +698,35 @@ const Tree = ({
   );
 };
 
+const filterTree = ({
+  text,
+  tree,
+}: {
+  text: string;
+  tree: HierarchyValue[];
+}) => {
+  const lowercaseText = text.toLowerCase();
+
+  const filter = (result: HierarchyValue[], d: HierarchyValue) => {
+    if (d.label.toLowerCase().includes(lowercaseText)) {
+      result.push(d);
+      return result;
+    }
+
+    if (Array.isArray(d.children)) {
+      const children = d.children.reduce(filter, []);
+
+      if (children.length > 0) {
+        result.push({ ...d, children });
+      }
+    }
+
+    return result;
+  };
+
+  return tree.reduce(filter, []);
+};
```

Comment:
1. Nit: Seems like it is similar to pruneTree, maybe it could be reused ?
2. Nit: I think it'd be clearer to use Array.filter instead of Array.reduce ?